bundles/org.openhab.binding.atlona/src/main/java/org/openhab/binding/atlona/internal/discovery/AtlonaDiscoveryParticipant.java

```diff
@@ -0,0 +1,128 @@
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Set.of(THING_TYPE_PRO3_44M, THING_TYPE_PRO3_66M, THING_TYPE_PRO3_88M, THING_TYPE_PRO3_1616M,
+                THING_TYPE_PRO3HD_44M, THING_TYPE_PRO3HD_66M);
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(SddpDevice device) {
+        final ThingUID uid = getThingUID(device);
+        if (uid != null) {
+            final Map<String, Object> properties = new HashMap<>(2);
+            final String label = device.model + " (" + device.ipAddress + ")";
+
+            properties.put(Thing.PROPERTY_MAC_ADDRESS, uid.getId());
+            properties.put(AtlonaPro3Config.IP_ADDRESS, device.ipAddress);
+
+            final DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                    .withRepresentationProperty(Thing.PROPERTY_MAC_ADDRESS).withLabel(label).build();
+
+            logger.debug("Created a DiscoveryResult for device '{}' with UID '{}'", label, uid.getId());
+            return result;
+        } else {
+            return null;
+        }
+    }
+
+    /*
+     * The Atlona SDDP message has the following format
+     *
+     * <pre>
+     * NOTIFY ALIVE SDDP/1.0
+     * From: "192.168.1.30:1902"
+     * Host: "AT-UHD-PRO3-88M_B898B0030F4D"
+     * Type: "AT-UHD-PRO3-88M"
+     * Max-Age: 1800
+     * Primary-Proxy: "avswitch"
+     * Proxies: "avswitch"
+     * Manufacturer: "Atlona"
+     * Model: "AT-UHD-PRO3-88M"
+     * Driver: "avswitch_Atlona_AT-UHD-PRO3-88M_IP.c4i"
+     * Config-URL: "http://192.168.1.30/"
+     * </pre>
+     */
+    @Override
+    public @Nullable ThingUID getThingUID(SddpDevice device) {
+        if (ATLONA.equals(device.manufacturer) && PROXY_AVSWITCH.equals(device.primaryProxy)
+                && !device.macAddress.isBlank()) {
+            final ThingTypeUID typeId;
+
+            switch (device.model) {
+                case "AT-UHD-PRO3-44M":
+                    typeId = THING_TYPE_PRO3_44M;
+                    break;
+                case "AT-UHD-PRO3-66M":
+                    typeId = THING_TYPE_PRO3_66M;
+                    break;
+                case "AT-UHD-PRO3-88M":
+                    typeId = THING_TYPE_PRO3_88M;
+                    break;
+                case "AT-UHD-PRO3-1616M":
+                    typeId = THING_TYPE_PRO3_1616M;
+                    break;
+                case "AT-PRO3HD44M":
+                    typeId = THING_TYPE_PRO3HD_44M;
+                    break;
+                case "AT-PRO3HD66M":
+                    typeId = THING_TYPE_PRO3HD_66M;
+                    break;
+                default:
+                    logger.warn("Unknown model #: {}", device.model);
+                    return null;
+            }
+
+            logger.debug("Atlona matrix with mac {} found at {}", device.macAddress, device.ipAddress);
+            return new ThingUID(typeId, device.macAddress);
```

Comment:
rather use `device.macAddress.replaceAll("[^A-Za-z0-9_]", "_")` as thing id to avoid unexpected characters.

Reply:
> use device.macAddress.replaceAll("[^A-Za-z0-9_]", "_") as thing id to avoid unexpected characters

@lolodomo two things..

1. your Regex is wrong. The dash `-` character IS an allowed character in the sub-parts of a ThingUID. So the exclusion Regex can safely include `- characters too.

2. As a separate PR I will submit a change to OH Core for the JavaDoc and ReadMe for SddpDevice to make it explicitly clear the contract that OH Core SddpDevice will fulfil for its returned values.